build/patches/Add-a-proxy-configuration-page-fix-377-679.patch

```diff
@@ -0,0 +1,236 @@
+       proxyConfig.proxy_rules().type = net::ProxyConfig::ProxyRules::Type::EMPTY;
++  } else if (t == "none") {
++      OnClear(nullptr);
++      return;
+   } else {
+      // invalid type
+      LOG(WARNING) << "invalid proxy configuration type";
+@@ -367,20 +382,23 @@ void ProxyConfigMessageHandler::apply(const net::ProxyConfig& proxyConfig) {
+   if (proxyConfig.auto_detect()) {
+     const base::Value cfg = ProxyConfigDictionary::CreateAutoDetect();
+     profile_->GetPrefs()->Set(proxy_config::prefs::kProxy, cfg);
+-    return;
+   } else if (proxyConfig.has_pac_url()) {
+     const base::Value cfg = ProxyConfigDictionary::CreatePacScript(proxyConfig.pac_url().spec(), proxyConfig.pac_mandatory());
+     profile_->GetPrefs()->Set(proxy_config::prefs::kProxy, cfg);
+-    return;
+-  }
+-
+-  auto proxyRulesAsString = proxyConfig.proxy_rules().ToString();
+-  auto bypassRulesAsString = proxyConfig.proxy_rules().bypass_rules.ToString();
++  } else if (proxyConfig.proxy_rules().type == net::ProxyConfig::ProxyRules::Type::EMPTY) {
++    const base::Value cfg = ProxyConfigDictionary::CreateDirect();
++    profile_->GetPrefs()->Set(proxy_config::prefs::kProxy, cfg);
++  } else {
++    auto proxyRulesAsString = proxyConfig.proxy_rules().ToString();
++    auto bypassRulesAsString = proxyConfig.proxy_rules().bypass_rules.ToString();
+ 
+-  // fixed servers
+-  const base::Value cfg = ProxyConfigDictionary::CreateFixedServers(proxyRulesAsString,
+-                          bypassRulesAsString, proxyConfig.proxy_rules().reverse_bypass);
+-  profile_->GetPrefs()->Set(proxy_config::prefs::kProxy, cfg);
++    // fixed servers
++    const base::Value cfg = ProxyConfigDictionary::CreateFixedServers(proxyRulesAsString,
++                            bypassRulesAsString, proxyConfig.proxy_rules().reverse_bypass);
++    profile_->GetPrefs()->Set(proxy_config::prefs::kProxy, cfg);
++  }
++  profile_->GetPrefs()->CommitPendingWrite();
++  OnEnableNotifyUIWithState(nullptr);
+ }
+ 
+ }  // namespace
+diff --git a/net/proxy_resolution/proxy_config.cc b/net/proxy_resolution/proxy_config.cc
+old mode 100644
+new mode 100755
+index 133ff6c46cfcb53938671c6ec4a22b663d00bd29..82cda2455b15a550f805d5620613185b075c5f1d
+--- a/net/proxy_resolution/proxy_config.cc
++++ b/net/proxy_resolution/proxy_config.cc
+@@ -110,7 +110,8 @@ void ProxyConfig::ProxyRules::ParseFromString(const std::string& proxy_rules) {
+                                    &single_proxies,
+                                    ProxyServer::SCHEME_HTTP);
+         type = Type::PROXY_LIST;
+-        return;
++        // bromite #679
++        continue; // was: return; maybe an upstream bug
+       }
+ 
+       // Trim whitespace off the url scheme.
+@@ -148,11 +149,16 @@ std::string ProxyConfig::ProxyRules::ToString() const {
+ 
+   // special case: a single proxy server specified
+   if (type == Type::PROXY_LIST) {
+-    if (single_proxies.size() == 1) {
+-      return single_proxies.Get().ToURI();
```

Comment:
This is where #679 was.

Reply:
yes, this value is what is passed to the prefs.
above is where it parses from prefs string